src/BlockTypes/AbstractBlock.php

```diff
@@ -335,6 +335,20 @@ protected function enqueue_data( array $attributes = [] ) {
 				$this->asset_data_registry->add( $asset_data_key, $asset_data_value );
 			}
 		}
+
+		if ( ! $this->asset_data_registry->exists( 'wordCountType' ) ) {
+			/*
+			* translators: If your word count is based on single characters (e.g. East Asian characters),
+			* enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+			* Do not translate into your own language.
+			*/
+			$this->asset_data_registry->add( 'wordCountType', _x( 'words', 'Word count type. Do not translate!', 'woo-gutenberg-products-block' ), true );
+		}
+
+		if ( ! $this->asset_data_registry->exists( 'productCount' ) ) {
+			$product_counts = wp_count_posts( 'product' );
+			$this->asset_data_registry->add( 'productCount', array_sum( (array) $product_counts ) );
+		}
```

Comment:
If you move `pluginUrl` and `buildPhase` into here, you could have `productCount` and `wordCountType` be a part of the `wcBlocksConfig` object too. Then they are appropriately namespaced and any client consumer just taking a look at the `wcSettings` global will see these are properties of `wcSettings.wcBlocksConfig` which makes it clearer they aren't a core config.